core/src/main/java/org/opensearch/sql/catalog/CatalogService.java

```diff
@@ -0,0 +1,15 @@
+package org.opensearch.sql.catalog;
+
+import java.util.Optional;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.storage.StorageEngine;
+
+
+public interface CatalogService {
+
+
+  Optional<StorageEngine> getStorageEngine(String catalog);
+  
+  Optional<ExecutionEngine> getExecutionEngine(String catalog);
```

Comment:
Just feel a little strange at first sight. Why different catalog has impact on execution engine choice?

Reply:
Good point. Ideally execution engine should be one and it should invoke relevant datastore from the physical plan provided. But in the current state, we have defined OpenSearchExecutionEngine and defined this in opensearch folder. I need to take this out of the opensearch folder and implement a common one which invokes both prometheus and opensearch. This way it can be extensible to future use cases for cross cutting concerns. For now, I am taking the shortest path having separate execution engines for both opensearch and prometheus. 


What can I do now is remove the above method from the CatalogService but have a public method in PPLCatalogService to serve the purpose temporarily.